### ISSUE TITLE
New Rule: Check import paths work on case sensitive file systems

### ISSUE THREAD

**Issue by tohagan** [eslint/eslint#16903]
### Rule details

Checks the import paths will work on case sensitive file systems 

### Related ECMAScript feature

import 

### What type of rule is this?

Warns about a potential problem

### Example code

```js
A source file name "MyCode.ts" or "MyCode.js" can be imported/required as "mycode" on Windows.
This compiles and runs on Windows NTFS but the import fails when deployed to a Unix/Linux or MacOS filesystem.

An eslint rule could checks that the case sensitive name stored in directories in a path matches the path used in an import or require statement.
```

### Why should this rule be in the core instead of a plugin?

This is a common and widely encountered coding issue for all Windows developers.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by tohagan** [comment]
I encountered this specific bug today. ESLint could have saved me.

**Comment by mdjermanovic** [comment]
Hi @tohagan, thanks for the issue!

ESLint doesn't include environment-specific rules in the core.

For ES imports, you can use plugin rule `import/no-unresolved` with option `caseSensitive: true` (default):

https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-unresolved.md

For `require`, you can use plugin rule `n/no-missing-require`:

https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-missing-require.md
